Java/cryptography/des/DESController.java

```diff
@@ -0,0 +1,153 @@
+package des;
+
+import des.exception.ValidationException;
+import des.key.KeyGenerationModels;
+
+/**
+ * <h1>Data Encryption Standard</h1>
+ * <p>
+ * 		To Implement Data Encryption Standard Symmetric Key Cryptography algorithm with 64-bit plain text and 64-bit cipher
+ * 		text
+ * </p>
+ * <b>Description</b>
+ * <p>
+ * 		class DESController is a public class which acts as a controller, it is used to call cryptography methods of
+ * 		DESCryptography class, the DES algorithm is the most widely used encryption algorithm in the world, DES has been
+ * 		found vulnerable/breakable against many attacks, reflecting it's a decline in popularity but despite this DES will
+ * 		live in many government and banking for years, due to its successor algorithms Triple-DES, AES, etc.
+ * </p>
+ * <b>Reference for understanding the algorithm</b>
+ * <p>
+ * 		<a href="http://page.math.tu-berlin.de/~kant/teaching/hess/krypto-ws2006/des.htm"> The DES Algorithm Illustrated </a>
+ * </p>
+ */
+public class DESController
+{
+	/**
+	 * This is the main method which makes use of the following methods:
+	 * <ul>
+	 * 		<li>DESCryptographyLogic.encrypt</li>
+	 * 		<li>DESCryptographyLogic.decrypt</li>
+	 * </ul>
+	 * 
+	 * And uses the following objects of types:
+	 * <ul>
+	 * 		<li>DESCryptographyLogic</li>
+	 * </ul>
+	 * 
+	 * @param args not used
+	 */
+	public static void main(String[] args)
+	{
+		// Defining plain text of 64-bits
+		byte[] plainText = {
+				1, 1, 1, 0, 0, 0, 0, 1,
+				0, 0, 1, 0, 0, 0, 1, 1,
+				0, 1, 0, 0, 0, 1, 0, 1,
+				0, 1, 1, 0, 0, 1, 1, 1,
+				1, 0, 0, 0, 1, 0, 0, 1,
+				1, 0, 1, 0, 1, 0, 1, 1,
+				1, 1, 0, 0, 1, 1, 0, 1,
+				1, 1, 1, 0, 1, 0, 1, 1
+		};
+		
+		// Defining original key of 64-bits
+		byte[] key = {
+				0, 0, 0, 1, 0, 0, 1, 1,
+				0, 0, 1, 1, 0, 1, 0, 0,
+				0, 1, 0, 1, 0, 1, 1, 1, 
+				0, 1, 1, 1, 1, 0, 0, 1,
+				1, 0, 0, 1, 1, 0, 1, 1,
+				1, 0, 1, 1, 1, 1, 0, 0,
+				1, 1, 0, 1, 1, 1, 1, 1,
+				1, 1, 1, 1, 0, 0, 0, 1
+		};
+		
+		//created object of type DESCryptographyModels to access models associated with the respective
+		DESCryptographyModels	desCryptographyModels	= new DESCryptographyModels();
+		//created object of type KeyGenerationModels to access models associated with the respective
+		KeyGenerationModels		keyGenerationModels		= new KeyGenerationModels();
+		
+		//setting plain text
+		desCryptographyModels.setPlainText(plainText);
+		//setting original key
+		desCryptographyModels.setKey(key);
+		
+		//printing plain text
+		System.out.println("Plain Text:");
+		for (int i = 0; i < plainText.length; i++)
+		{
+			//used for styling the output
+			if (i % 8 == 0 && i != 0)
+			{

```

Comment:
Can the Core Logic of the Code shift into some other function? We should ideally use the main() function only for the Driver Code

Reply:
Okay I understand your concern I'll do that for sure, really thank you for guidance.